pkg/metrics/prometheus_parser.go

```diff
@@ -0,0 +1,155 @@
+package metrics
+
+import (
+	"bufio"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// PrometheusMetric represents a single Prometheus metric
+type PrometheusMetric struct {
+	Name   string
+	Labels map[string]string
+	Value  string
+	Help   string
+	Type   string
+}
+
+// PrometheusParser parses Prometheus text format metrics
+type PrometheusParser struct {
+	commentRegex *regexp.Regexp
+	metricRegex  *regexp.Regexp
+}
+
+// NewPrometheusParser creates a new Prometheus metrics parser
+func NewPrometheusParser() *PrometheusParser {
```

Comment:
Can't we use https://pkg.go.dev/github.com/prometheus/prometheus/promql/parser?

Reply:
I get a 500 when visit the link but I assume its temporary. I'll take a look tomorrow, thanks!

Reply:
I've changed to use [https://pkg.go.dev/github.com/prometheus/common/expfmt]( https://pkg.go.dev/github.com/prometheus/common/expfmt) so I can use `families, err := parser.TextToMetricFamilies(strings.NewReader(string(body)))`
And then I can get metrics per family and add our labels. 
Let me know what do you think 🙏